shell/components/nav/WindowManager/ContainerShell.vue

```diff
@@ -220,11 +215,11 @@ export default {
         return;
       }
 
-      const { node } = this;
-      let cmd = DEFAULT_COMMAND;
-
-      if (!isEmpty(node) && node?.status?.nodeInfo?.operatingSystem === 'windows') {
-        cmd = ['cmd'];
+      if (this.pod.os) {
+        this.os = this.pod.os;
+        this.backupShells = this.backupShells.filter((shell) => shell !== this.pod.os);
+      } else {
+        this.os = this.backupShells.shift();
```

Comment:
Could using `shift` here mutate `this.backupShells` by removing the first element from the array? Could this be an undesired side-effect?

Reply:
It's entirely on purpose, when I assign the OS to the data prop I also want to remove it from backup shells so we don't end up using it again and ending up in an endless loop.